### PR TITLE
NEP-5 balanceOf validation

### PR DESCRIPTION
**Pull request by brianlenz**
* Added proper validation to the balanceOf function so that the function only works with proper addresses as opposed to any arbitrary argument that could read any data from the storage.

### CODE REVIEW

**Review by localhuman — APPROVED** [review]
This looks good!  Though I don't believe it closes any security issue.  Anyone can retrieve any value of storage by key when using the RPC `getstorage` method.

**Comment by brianlenz** [comment]
Agreed it’s not a security issue! All of the data is on the blockchain, after all. It just addresses the expected, functional behavior for “balanceOf”.

FWIW, this was identified via our security audit (low risk, of course)...

File: nex/token/nep5.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ def handle_nep51(self, operation, args, token: Token):
         elif operation == 'balanceOf':
             if len(args) == 1:
                 account = args[0]
+                if len(account) != 20:
```
Comment (metachris) [inline nex/token/nep5.py]: Isn't a typical address length 34?

**Comment by brianlenz** [comment]
It’s 34 in Base58Check, but I believe the addresses in the AVM runtime should be 160-bit == 20 bytes, I thought?

File: nex/token/nep5.py (hunk 1 above)
Comment (brianlenz) [reply nex/token/nep5.py]: It’s 34 in Base58Check, but I believe the addresses in the AVM runtime should be 160-bit == 20 bytes, I thought?

**Comment by shargon** [comment]
Is not a security issue, but maybe could be inside good practice to accustom people to check the parameters received, it does not affect safety, but we think it is good to do the checking.

Imagine the following scenario. An application rewards or checks that the address of a user's profile has a balance. However, the address has no balance, but the user instead of a valid address (already reserved by other profiles) sets an arbitrary value that returns a value with balance correctly.

**Pull request closed without merging**